`libraspberrypi` fails to build on `nixos-unstable`

Hey!

I was trying to build myself a custom SD card today and noticed that `libraspberrypi` isn't building on `nixos-unstable`:

```
error: build of '/nix/store/216kczsf5wjy1cj68g7p9d2s588wzdkh-libraspberrypi-unstable-2024-12-23.drv' on 'ssh-ng://builder@linux-builder' failed: Cannot build '/nix/store/216kczsf5wjy1cj68g7p9d2s588wzdkh-libraspberrypi-unstable-2024-12-23.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/29gyl25w1hdmmlrm89r1rifmv0lhccy5-libraspberrypi-unstable-2024-12-23
       Last 25 log lines:
       >       |                                                                                                                 ^~~
       > /build/source/interface/vchiq_arm/vchiq_lib.c:1551:13: note: in expansion of macro 'vcos_log_trace'
       >  1551 |             vcos_log_trace( "callback(%x, %x, %x(%x,%x), %x)",
       >       |             ^~~~~~~~~~~~~~
       > /build/source/interface/vchiq_arm/vchiq_lib.c:1553:73: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
       >  1553 |                (uint32_t)&service->base, (uint32_t)service->lib_handle, (uint32_t)service->base.userdata, (uint32_t)completion->bulk_userdata );
       >       |                                                                         ^
       > /build/source/build/inc/interface/vcos/vcos_logging.h:234:113: note: in definition of macro '_VCOS_LOG_X'
       >   234 | #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
       >       |                                                                                                                 ^~~
       > /build/source/interface/vchiq_arm/vchiq_lib.c:1551:13: note: in expansion of macro 'vcos_log_trace'
       >  1551 |             vcos_log_trace( "callback(%x, %x, %x(%x,%x), %x)",
       >       |             ^~~~~~~~~~~~~~
       > /build/source/interface/vchiq_arm/vchiq_lib.c:1553:107: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
       >  1553 |                (uint32_t)&service->base, (uint32_t)service->lib_handle, (uint32_t)service->base.userdata, (uint32_t)completion->bulk_userdata );
       >       |                                                                                                           ^
       > /build/source/build/inc/interface/vcos/vcos_logging.h:234:113: note: in definition of macro '_VCOS_LOG_X'
       >   234 | #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
       >       |                                                                                                                 ^~~
       > /build/source/interface/vchiq_arm/vchiq_lib.c:1551:13: note: in expansion of macro 'vcos_log_trace'
       >  1551 |             vcos_log_trace( "callback(%x, %x, %x(%x,%x), %x)",
       >       |             ^~~~~~~~~~~~~~
       > make[2]: *** [interface/vchiq_arm/CMakeFiles/vchiq_arm.dir/build.make:79: interface/vchiq_arm/CMakeFiles/vchiq_arm.dir/vchiq_lib.c.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:490: interface/vchiq_arm/CMakeFiles/vchiq_arm.dir/all] Error 2
       > make: *** [Makefile:156: all] Error 2
       For full logs, run:
         nix-store -l /nix/store/216kczsf5wjy1cj68g7p9d2s588wzdkh-libraspberrypi-unstable-2024-12-23.drv
error: Cannot build '/nix/store/216kczsf5wjy1cj68g7p9d2s588wzdkh-libraspberrypi-unstable-2024-12-23.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/29gyl25w1hdmmlrm89r1rifmv0lhccy5-libraspberrypi-unstable-2024-12-23
error: Cannot build '/nix/store/fykb8yhkl6mjnsrswgqh11vw0906vry0-raspberrypi-eeprom-2026.01.09-2711.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/9xwzb2ga702778rqvnavja6gwj0wyhra-raspberrypi-eeprom-2026.01.09-2711
error: Cannot build '/nix/store/xkwnqy1sg921zgf4nvwb5nvhf6s0wckd-man-paths.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/d4pf71zqpw84s3xqabvkanfwcp67mghw-man-paths
error: Cannot build '/nix/store/3p4qjklzcwwclvsigqvngg4cry108758-raspberrypi-eeprom-2026.01.09-2711_fish-completions.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/gfqchxh3anadcjagdpd8i600rwqldkwr-raspberrypi-eeprom-2026.01.09-2711_fish-completions
error: Cannot build '/nix/store/pc0jvv8nrwfhb2w8ng4a3i01v9k09b94-system-path.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/hxsvcny567jlpfvcay6qwbqbqzc1q64l-system-path
error: Cannot build '/nix/store/2r4h6b0cv41lihszwgf0pqzs3jmxjg26-nixos-system-rpi5-sd-card-26.05.20260121.88d3861.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/x7qnwdjybqbh049w1pxfq2s23v1gcxqh-nixos-system-rpi5-sd-card-26.05.20260121.88d3861
error: Cannot build '/nix/store/jxi5knaqqynn3di56xv8hx9lkyblpzz3-nixos-image-rpi5-kernel.img.zst.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/gx3q2ngcxdljb77fcmrjmza232p35i8q-nixos-image-rpi5-kernel.img.zst
```

I made a (really stupid) fork that just comments out the `libraspberrypi` dep from the flake and vendor pkg overlay since `libraspberrypi` appears to be an antique by this point.

My fork, in case anyone else comes across this issue and just wants to get something working fast: https://github.com/GregoryConrad/nixos-raspberrypi

Obviously, my fix isn't upstream-able in its current state. How can we fix this for everyone else?

--- a/pkgs/raspberrypi/libraspberrypi.nix
+++ b/pkgs/raspberrypi/libraspberrypi.nix
@@ -16,6 +16,7 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
   nativeBuildInputs = [ cmake pkg-config ];
+  env.NIX_CFLAGS_COMPILE = "-std=gnu17";
   cmakeFlags = [
     # -DARM64=ON disables all targets that only build on 32-bit ARM; this allows
     # the package to build on aarch64 and other architectures

